spec/ic.did

```diff
@@ -16,6 +16,20 @@ type definite_canister_settings = record {
   freezing_threshold : nat;
 };
 
+type http_header = record { 0: text; 1: text };
+
+type http_response = record {
+  status: nat;
+  headers: vec http_header;
+  body: blob;
+};
+
+type http_request_error = variant {
+  no_consensus;
+  timeout;
+  bad_tls;
```

Comment:
Reading the spec, I think we need a couple of more errors:

```
malformed_url; // in case the url isn't valid
not_https; // url is valid, but isn't https
```

Both of these cases can arguably be merged into a single error as well. No strong opinions on my side.